test/unit/analysis/epacamd_eia_test.py

```diff
@@ -0,0 +1,198 @@
+        mock_cems_extended["operating_datetime_utc"] == "2019-12-31 22:00:00+00:00"
+    ]
+
+    # simulate join by duplicating rows as appropriate
+    one_to_many = uniques.query('plant_id_eia == 12 and emissions_unit_id_epa == "a"')
+    many_to_many = uniques.query('plant_id_eia == 10 and emissions_unit_id_epa == "c"')
+    expected = (
+        pd.concat([uniques, one_to_many, many_to_many])
+        .sort_index()
+        .reset_index(drop=True)
+    )
+
+    expected = expected.assign(
+        plant_id_eia=expected["plant_id_eia"],
+        emissions_unit_id_epa=expected["emissions_unit_id_epa"],
+        generator_id=[0, 1, 1, 2, 0, 0, 0, 1],
+        global_subplant_id=[0, 1, 1, 1, 2, 2, 3, 3],
+        subplant_id=[0, 1, 1, 1, 0, 0, 0, 0],
+    )
+
+    input_ = expected.drop(columns=["subplant_id"])
+    actual = cw_analysis._convert_global_id_to_composite_id(input_)
+    assert_frame_equal(actual, expected)
+
+
+def test_make_subplant_ids(mock_crosswalk, mock_cems_extended):
+    """Integration test for the subplant_id assignment process.
+
+    The new subplant_id column is half of the compound key (plant_id_eia, subplant_id)
+    that should identify disjoint subgraphs of units and generators.
+
+       subplant_id  ...  plant_id_eia  emissions_unit_id_epa  generator_id
+    0            0  ...        10            a                 0          # one to one
+    1            1  ...        10            b                 1          # many to many
+    2            1  ...        10            c                 1
+    3            1  ...        10            c                 2
+    4            0  ...        11            a                 0          # one to many
+    5            0  ...        11            b                 0
+    6            0  ...        12            a                 0          # many to one
+    7            0  ...        12            a                 1
+
+    """
+    uniques = mock_cems_extended[
+        mock_cems_extended["operating_datetime_utc"] == "2019-12-31 22:00:00+00:00"
+    ]
+
+    # simulate join by duplicating rows as appropriate
+    one_to_many = uniques.query('plant_id_eia == 12 and emissions_unit_id_epa == "a"')
+    many_to_many = uniques.query('plant_id_eia == 10 and emissions_unit_id_epa == "c"')
+    expected = (
+        pd.concat([uniques, one_to_many, many_to_many])
+        .sort_index()
+        .reset_index(drop=True)
+    )
+
+    expected = expected.assign(
+        plant_id_eia=expected["plant_id_eia"],
+        emissions_unit_id_epa=expected["emissions_unit_id_epa"],
+        generator_id=[0, 1, 1, 2, 0, 0, 0, 1],
+        subplant_id=[0, 1, 1, 1, 0, 0, 0, 0],
+    )
+    # fix column order and drop some columns
+    expected = expected[
+        ["subplant_id", "plant_id_eia", "emissions_unit_id_epa", "generator_id"]
+    ]
+
+    # should be two separate tests but I ran out of time
+    actual1 = cw_analysis.filter_crosswalk(mock_crosswalk, mock_cems_extended)
```

Comment:
what is the corresponding `expected1`? I'd say either add that and test it or rename `actual1` so it doesn't give that false impression that it is being tested (at least that's what I think when I see the "actual" variable in a test)